cmd/stellar-rpc/internal/methods/get_events.go

```diff
@@ -142,6 +128,21 @@ func (g *GetEventsRequest) Valid(maxLimit uint) error {
 		}
 	}
 
+	if g.Pagination != nil { //nolint:nestif
+		if g.Pagination.Cursor != nil && (g.StartLedger != 0 || g.EndLedger != 0) {
+			return errors.New("ledger ranges and cursor cannot both be set")
+		}
+		if g.Pagination.Limit > maxLimit {
+			return fmt.Errorf("limit must not exceed %d", maxLimit)
+		}
+	}
+	if g.StartLedger <= 0 {
+		return errors.New("startLedger must be positive")
+	}
```

Comment:
Now this will fail whenever `pagination` is set by itself, since `startLedger == 0` in that case, right? lol. This tells me we need better unit testing on this validation path.

Reply:
I have restored the original validation logic. I have actually realized that not setting the end of the range means _unlimited_, so we shouldn't be validating that end >= start (since end could be 0).